health.go

```diff
@@ -92,15 +107,15 @@ func (h *health) stopCheckTask(name string) {
 	delete(h.checkTasks, name)
 }
 
-func (h *health) scheduleCheck(task *checkTask, cfg *Config) {
+func (h *health) scheduleCheck(task *checkTask, initialDelay time.Duration, executionPeriod time.Duration) {
```

Comment:
I don't like this change. It doesn't improve the code, and it actually makes it more prone to (internal) API changes

Reply:
> It doesn't improve the code

Well, that's debatable. It removes an extra level of indirection AND it reduces the dependencies (this method doesn't need everything from the config). And the config structure is actually dropped, because it's not needed anymore.

Reply:
It's only more readable when you make the arguments more "strongly typed". When you have several arguments of the same type the API is confusing to use. If you create types for the args, it becomes very verbose and annoying

I think it's more likely that the function will require more arguments, than having a change in the config struct that will break the compilation.

In this case I think it's better to leave it as is

Reply:
I disagree, but I made the requested changes.